### ISSUE TITLE
AttributeError when saving belongs_to relation

### ISSUE THREAD

**Issue by joaobarbosa** [sdispater/orator#120]
Hi there,

I'm having an issue when trying to save a model with a ```@belongs_to``` field. The model:

```python
from orator import Model
from orator.orm import belongs_to

class Device(Model):
    __table__ = 'device'

    @belongs_to('product_id', 'id')
    def product(self):
        from .product import Product
        return Product
```

How I'm trying to save it:

```python
product = Product.find(1)

device = Device()
device.device_token = payload['device_identifier']
device.service_token = payload['service_device_id']
device.model = payload['device_model']
device.os = payload['device_os']
device.product().associate(product)  # Raises exception
# device.product_id = 1 -- This works!
```

Exception:

```
Traceback (most recent call last):
  ...
  File "/project_path/blueprints/api/push.py", line 55, in post
    device.product().associate(product)
  File "/env_path/orator/orm/model.py", line 2816, in __getattr__
    return self.get_attribute(item)
  File "/env_path/orator/orm/model.py", line 2292, in get_attribute
    relation = original or super(Model, self).__getattribute__(key)
  File "/env_path/orator/orm/utils.py", line 184, in __get__
    relation = self._get(instance)
  File "/env_path/orator/orm/utils.py", line 290, in _get
    _wrapped=False
  File "/env_path/orator/orm/model.py", line 783, in belongs_to
    rel = BelongsTo(query, self, foreign_key, other_key, relation)
  File "/env_path/orator/orm/relations/belongs_to.py", line 31, in __init__
    super(BelongsTo, self).__init__(query, parent)
  File "/env_path/orator/orm/relations/relation.py", line 26, in __init__
    self.add_constraints()
  File "/env_path/orator/orm/relations/belongs_to.py", line 48, in add_constraints
    self._query.where('%s.%s' % (table, self._other_key), '=', getattr(self._parent, self._foreign_key))
  File "/env_path/orator/orm/model.py", line 2816, in __getattr__
    return self.get_attribute(item)
  File "/env_path/orator/orm/model.py", line 2292, in get_attribute
    relation = original or super(Model, self).__getattribute__(key)
AttributeError: 'Device' object has no attribute 'product_id'
```

But, the table has the column 'product_id', because I can set it directly using ```device.product_id = x```. Is there anything wrong?

Thanks!

**Comment by sdispater** [comment]
Thanks for reporting this. I will investigate it and get back to you.

**Comment by joaobarbosa** [comment]
It's happening with ```.attach()``` (```belongs_to_many``` relationship) also:

```python
class Device(Model):
    # ...

    @belongs_to_many('devices_users', 'device_id', 'user_id')
    def users(self):
        from .user import User
        return User
```

```python
user = User.find(1)

device = Device()
# ...
device.users().attach(user)  # Raises exception
```

```
AttributeError: 'Device' object has no attribute 'id'
```

**Comment by joaobarbosa** [comment]
Figured out what was going on with ```.attach()```: I was trying to attach an model to another before saving the Device object! Saving the model before performing the ```.attach()``` solved it (but is this expected? I thought it would create the relationship after saving it into the database).

```python
user = User.find(1)

device = Device()
# ...
device.save() -- now it has an id!
device.users().attach(user)  # Raises exception
```

I've tried to apply the same workaround for the ```.associate()``` issue (setting the relation field to be ```nullable``` to allow saving the model before associating), but the same error raised.

**Comment by sdispater** [comment]
Yeah. `associate()` and `attach()` will only work with a saved object because when you call `users()`, Orator will try to get the foreign key on the object.

The best approach for now would be to save the relation the other way:

```python
product = Product.find(1)

device.device_token = payload['device_identifier']
device.service_token = payload['service_device_id']
device.model = payload['device_model']
device.os = payload['device_os']

product.devices().save(device)
```

But, I will try to improve the way `belongs_to` relations are handled.

**Comment by sdispater** [comment]
I just pushed a fix for the `associate()` method (c441473a9a415a981df6c50fd45a88c616220fff) to the `0.9` branch.

Regarding `attach()` I will not change the behavior since you can only attach  two existing models.

**Comment by joaobarbosa** [comment]
Thanks a lot, @sdispater!
